corehq/apps/app_manager/models.py

```diff
@@ -1218,6 +1219,18 @@ def render_xform(self, build_profile_id=None):
         self.add_stuff_to_xform(xform, build_profile_id)
         return xform.render()
 
+    def cached_get_questions(self, include_triggers=False, include_groups=False):
+        # call to get_questions with a superset of necessary information, so
+        # it can hit the same cache across common app-building workflows
+        # performs filtering so it's a drop-in replacement for get_questions
+        def matches_params(question):
+            return (
+                (include_triggers or not question.get('is_trigger'))
+                and (include_groups or not question.get('is_group'))
+            )
+        questions = self.get_questions([], include_triggers=True, include_groups=True)
+        return [q for q in questions if matches_params(q)]
+
```

Comment:
I also played around with making `cached_get_questions` not take _any_ parameters, which I kinda liked.  Then the call that doesn't want groups would do something like:
```python
has_questions = any(not q.get('is_group') for q in questions)
```
That would simplify this method _and_ I think make the reason for that distinction more obvious.  I ultimately decided against it in the hopes that this would be a step towards moving other calls to `get_questions` towards a shared cache.  Could be persuaded otherwise though.

Reply:
> making `cached_get_questions` not take _any_ parameters

I like this idea. I think the question that would decide it for me would be "Would that make the call sights duplicate complicated logic?" If not, then I think it's a good idea. Is that persuasive enough to get you over the hump?

Reply:
At the moment, no, there would not be any duplicated logic.  However, most of the calls to `get_questions` either don't want triggers or don't want groups (or neither), so _if_ they get switched over, it'd make sense to put that filtering logic here.  That said, that's a hypothetical future world, so YAGNI suggests I should write it for the world we have now.